runtime-light/state/instance-state.cpp

```diff
@@ -42,8 +47,25 @@ int32_t merge_output_buffers() noexcept {
 
 void InstanceState::init_script_execution() noexcept {
   runtime_context.init();
-  init_php_scripts_in_each_worker(php_script_mutable_globals_singleton, main_task_);
-  scheduler.suspend(std::make_pair(main_task_.get_handle(), WaitEvent::Rechedule{}));
+  task_t<void> script_task;
+  init_php_scripts_in_each_worker(php_script_mutable_globals_singleton, script_task);
+
+  auto main_task{std::invoke(
+    [](task_t<void> script_task) noexcept -> task_t<void> {
+      // wrap script with additional check for unhandled exception
+      script_task = std::invoke(
+        [](task_t<void> script_task) noexcept -> task_t<void> {
+          co_await script_task;
+          if (auto exception{ForkInstanceState::get().current_info().get().thrown_exception}; !exception.is_null()) [[unlikely]] {
+            php_error("unhandled exception '%s' at %s:%" PRId64, exception.get_class(), exception->$file.c_str(), exception->$line);
+          }
+        },
+        std::move(script_task));
+      php_assert(co_await f$wait_concurrently(co_await start_fork_t{std::move(script_task)}));
+    },
+    std::move(script_task))};
+  scheduler.suspend(std::make_pair(main_task.get_handle(), WaitEvent::Rechedule{}));
```

Comment:
NIT: I think `{main_task.get_handle(), WaitEvent::Reschedule{}}` will be nicer. Maybe `std::pair` prefix will be needed.